pytest_mergify/quarantine.py

```diff
@@ -0,0 +1,79 @@
+import dataclasses
+import pytest
+import _pytest.nodes
+import os
+import requests
+import typing
+
+
+def get_branch_name_for_quarantine() -> typing.Optional[str]:
+    for env_var in ("MERGIFY_TESTS_TARGET_BRANCH", "GITHUB_BASE_REF", "GITHUB_REF"):
+        env_val = os.getenv(env_var)
+        if env_val is not None:
+            return env_val
+
+    return None
```

Comment:
This seems to be a way to get the base branch? There's some code in the tracer detector that is CI specific and does this, so I'm wondering if we shouldn't make this common. I'm afraid the tracer and quarantine have different behaviour in the long term.